File: src/Styles.fs
module internal Styles

open Fable.Helpers.ReactNative
open Fable.Helpers.ReactNative.Props

let [<Literal>] brandPrimary = "#428bca"
let [<Literal>] brandInfo = "#5bc0de"
let [<Literal>] brandSuccess = "#5cb85c"
let [<Literal>] brandDanger = "#d9534f"
let [<Literal>] brandWarning = "#f0ad4e"
let [<Literal>] brandSidebar = "#252932"

let [<Literal>] inverseTextColor = "#000"

let [<Literal>] textColor = "#FFFFFF"

let [<Literal>] shadowColor = "#000000"

let [<Literal>] backgroundColor = "#615A5B"
let [<Literal>] inputBackgroundColor = "#251D1C"

let [<Literal>] touched = "#5499C4"

let [<Literal>] fontSizeBase = 15.
let [<Literal>] smallFontSize = 10.
let [<Literal>] mediumFontSize = 12.
let [<Literal>] titleFontSize = 17.

let [<Literal>] borderRadius = 4.


let renderText fontSize =
    TextProperties.Style [ 
        TextStyle.Color textColor
        TextStyle.TextAlign TextAlignment.Center
        FlexStyle.Margin 3.
        TextStyle.FontSize fontSize
      ]

let defaultText<'a> = renderText fontSizeBase
let mediumText<'a> = renderText mediumFontSize
let smallText<'a> = renderText smallFontSize
let titleText<'a> = renderText titleFontSize

let whitespace<'a> = text [ smallText ] ""

let sceneBackground<'a> =
    ViewProperties.Style [ 
        FlexStyle.AlignSelf Alignment.Stretch
        FlexStyle.Padding 20.
        ViewStyle.ShadowColor shadowColor
        ViewStyle.ShadowOpacity 0.8
        ViewStyle.ShadowRadius 3.
        FlexStyle.JustifyContent JustifyContent.Center
        FlexStyle.Flex 1.
        ViewStyle.BackgroundColor backgroundColor
      ]

let viewPagerBackground<'a> =
    ViewPagerAndroidProperties.Style [
        FlexStyle.AlignSelf Alignment.Stretch
        FlexStyle.Padding 20.
        ViewStyle.ShadowColor shadowColor
        ViewStyle.ShadowOpacity 0.8
        ViewStyle.ShadowRadius 3.
        FlexStyle.JustifyContent JustifyContent.Center
        FlexStyle.Flex 1.
        ViewStyle.BackgroundColor backgroundColor
      ]
      
let button label onPress =
    button [
        ButtonProperties.Title label
        ButtonProperties.OnPress onPress
    ] [ ]

File: output/Nightwatch/Styles.js
import { Props_ViewStyle, Props_ViewProperties_Style_7FB9FF3D, Props_FlexStyle, Props_TextStyle, Props_TextProperties_Style_7FB9FF3D } from "./.fable/Fable.React.Native.3.0.0-alpha001/Fable.ReactNative.fs.js";
import { singleton, ofArray } from "./.fable/fable-library.3.0.0/List.js";
import * as react from "react";
import { Button, Text as Text$ } from "react-native";
import { keyValueList } from "./.fable/fable-library.3.0.0/MapUtil.js";

export function renderText(fontSize) {
    return Props_TextProperties_Style_7FB9FF3D(ofArray([new Props_TextStyle(0, "#FFFFFF"), new Props_TextStyle(7, "center"), new Props_FlexStyle(23, 3), new Props_TextStyle(2, fontSize)]));
}

export function defaultText() {
    return renderText(15);
}

export function mediumText() {
    return renderText(12);
}

export function smallText() {
    return renderText(10);
}

export function titleText() {
    return renderText(17);
}

export function whitespace() {
    const props = singleton(smallText());
    return react.createElement(Text$, keyValueList(props, 1), "");
}

export function sceneBackground() {
    return Props_ViewProperties_Style_7FB9FF3D(ofArray([new Props_FlexStyle(2, "stretch"), new Props_FlexStyle(37, 20), new Props_ViewStyle(25, "#000000"), new Props_ViewStyle(27, 0.8), new Props_ViewStyle(28, 3), new Props_FlexStyle(21, "center"), new Props_FlexStyle(14, 1), new Props_ViewStyle(1, "#615A5B")]));
}

export function button(label, onPress) {
    return react.createElement(Button, {
        title: label,
        onPress: onPress,
    });
}

